### ISSUE TITLE
CMake build fails with `"STREQUAL" "small" Unknown arguments specified`

### ISSUE THREAD

**Issue by Patater** [ARMmbed/mbed-os-example-for-google-iot-cloud#25]
### Description of defect

```
[EXAMPLES]> INFO     In folder 'mbed-os-example-for-google-iot-cloud'
[EXAMPLES]> INFO     Compiling mbed-os-example-for-google-iot-cloud DISCO_L475VG_IOT01A GCC_ARM
[EXAMPLES]> INFO     Executing command 'mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c'...

#### STDOUT ####
Configuring project and generating build system...
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python3: /usr/bin/python3.7 (found version "3.7.4") found components: Interpreter 
-- Checking for Python package prettytable -- found
-- Checking for Python package future -- found
-- Checking for Python package jinja2 -- found
-- Checking for Python package intelhex -- found
-- Configuring incomplete, errors occurred!
See also "/builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/cmake_build/DISCO_L475VG_IOT01A/develop/GCC_ARM/CMakeFiles/CMakeOutput.log".

#### STDERR ####
CMake Error at mbed-os/tools/cmake/app.cmake:10 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/.mbedbuild/mbed_config.cmake
Call Stack (most recent call first):
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/tools/cmake/toolchain.cmake:103 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/mbed-os/tools/cmake/toolchains/.cmake
Call Stack (most recent call first):
  mbed-os/tools/cmake/app.cmake:15 (include)
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/tools/cmake/toolchain.cmake:104 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/mbed-os/tools/cmake/cores/.cmake
Call Stack (most recent call first):
  mbed-os/tools/cmake/app.cmake:15 (include)
  CMakeLists.txt:10 (include)


CMake Error at mbed-os/CMakeLists.txt:8 (include):
  include could not find load file:

    /builds/workspace/mbed-os-ci_cmake-cloud-example-GCC_ARM-dev/cloud-examples/mbed-os-example-for-google-iot-cloud/.mbedbuild/mbed_config.cmake


CMake Error at mbed-os/CMakeLists.txt:29 (if):
  if given arguments:

    "STREQUAL" "small"

  Unknown arguments specified


ERROR: CMake invocation failed!

More information may be available by using the command line option '-v'.

#### End of STDOUT/STDERR ####
```

#### Target(s) affected by this defect ?
DISCO_L475VG_IOT01A

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM

#### What version of Mbed-os are you using (tag or sha) ?

9bbc8c914c7e 

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
Mbed CLI 2

#### How is this defect reproduced ?
`mbed-tools compile -t GCC_ARM -m DISCO_L475VG_IOT01A -c`

**Comment by ciarmcom** [comment]
Thank you for raising this detailed GitHub issue. I am now notifying our internal issue triagers. 
*Internal Jira reference: https://jira.arm.com/browse/IOTOSM-3797*

**Comment by LDong-Arm** [comment]
Fix merged (#26), @Patater @harmut01 please close the ticket (I don't have permissions), thanks
